src/node_url.cc

```diff
@@ -1597,10 +1597,13 @@ void URL::Parse(const char* input,
                    ch == '#' ||
                    special_back_slash) {
           if (buffer.size() > 0) {
-            int port = 0;
-            for (size_t i = 0; i < buffer.size(); i++)
+            unsigned port = 0;
+            for (size_t i = 0; i < buffer.size(); i++) {
               port = port * 10 + buffer[i] - '0';
-            if (port < 0 || port > 0xffff) {
+              // prevent integer overflow
+              if (port > 0xffff) break;
```

Comment:
nit: the `port > 0xffff` could be moved into the `for`-loop head

Reply:
Seems reasonable, moved.